src/js/node/fs.promises.ts

```diff
@@ -362,6 +362,9 @@ export default exports;
     }
 
     async read(buffer, offset, length, position) {
+      if (length === 0) {
```

Comment:
This is semantically different

This does not do a read() system call

If the file descriptor is invalid, a 0-length read would be EBADF, or if something else was wrong with the file, it would error rather than return 0

Reply:
True, for best results this fix should be in the zig code, but I dont feel comfortable making changes in zig

Preferably someone could make another PR that would close the issue 🙏